mungegithub/mungers/lgtm_handler.go

```diff
@@ -0,0 +1,172 @@
+	// Assumption: The comments should be sorted (by default from github api) from oldest to latest
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		fields := strings.Fields(*comment.Body)
+		if isCancelComment(fields) {
+			// "/lgtm cancell" if commented more recently than "/lgtm"
+			return
+		}
+		if isLGTMComment(fields) {
+			// TODO: support more complex policies for multiple reviewers.
+			// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+			// TODO: An approver should be acceptable.
+			// See https://github.com/kubernetes/contrib/pull/1428#discussion_r72563935
+			if isValidUser(comment.User) && isReviewer(comment.User, reviewers) {
+				glog.Infof("Adding lgtm label. Reviewer (%s) LGTM", *comment.User.Login)
+				obj.AddLabel(lgtmLabel)
+				return
+			}
+		}
+	}
+}
+
+func (h *LGTMHandler) removeLGTMIfCancelled(obj *github.MungeObject, comments []*githubapi.IssueComment, reviewers []*githubapi.User) {
+	for i := len(comments) - 1; i >= 0; i-- {
+		comment := comments[i]
+		fields := strings.Fields(*comment.Body)
+		if isLGTMComment(fields) {
+			// "/lgtm" if commented more recently than "/lgtm cancel"
+			return
+		}
+		if isCancelComment(fields) {
+			if isValidUser(comment.User) && isReviewer(comment.User, reviewers) {
+				glog.Infof("Removing lgtm label. Reviewer (%s) cancelled", *comment.User.Login)
+				obj.RemoveLabel(lgtmLabel)
+				return
+			}
+		}
+	}
+}
+
+func isLGTMComment(fields []string) bool {
+	return len(fields) == 1 && strings.ToLower(fields[0]) == "/lgtm"
+}
+
+func isCancelComment(fields []string) bool {
+	return len(fields) == 2 && strings.ToLower(fields[0]) == "/lgtm" && strings.ToLower(fields[1]) == "cancel"
+}
+
+func getReviewers(obj *github.MungeObject) []*githubapi.User {
+	// Note: assuming assignees are reviewers
+	allAssignees := append(obj.Issue.Assignees, obj.Issue.Assignee)
+	return filterValid(allAssignees)
+}
+
+func getComments(obj *github.MungeObject) ([]*githubapi.IssueComment, error) {
+	afterLastModified := func(opt *githubapi.IssueListCommentsOptions) *githubapi.IssueListCommentsOptions {
+		// Only comments updated at or after this time are returned.
+		// One possible case is that reviewer might "/lgtm" first, contributor updated PR, and reviewer updated "/lgtm".
+		// This is still valid. We don't recommend user to update it.
+		lastModified := *obj.LastModifiedTime()
+		opt.Since = lastModified
+		return opt
+	}
+	return obj.ListComments(afterLastModified)
+}
+
+func isValidUser(u *githubapi.User) bool {
+	return u != nil && u.Login != nil
+}
+
+func isReviewer(user *githubapi.User, reviewers []*githubapi.User) bool {
+	for _, r := range reviewers {
+		if !isValidUser(r) {
+			continue
+		}
+		if *user.Login == *r.Login {
+			return true
+		}
+	}
+	return false
+}
+
+// filterValid returns only user data that has valid username.
+func filterValid(users []*githubapi.User) (res []*githubapi.User) {
```

Comment:
Some of this logic was added to [mungegithub/mungers/mungerutil/util.go](https://github.com/apelisse/contrib/blob/4023b55ac693ce87dbe461bd3ba57c694c3950be/mungegithub/mungers/mungerutil/util.go#L31) as it is shared by multiple mungers. We could reuse that probably.